pkg/app/piped/executor/waitapproval/waitapproval.go

```diff
@@ -98,21 +104,15 @@ func (e *Executor) checkApproval(ctx context.Context) (string, bool) {
 		}
 	}
 	if approveCmd == nil {
-		return "", false
-	}
-
-	metadata := map[string]string{
-		approvedByKey: approveCmd.Commander,
-	}
-	if err := e.MetadataStore.Stage(e.Stage.Id).PutMulti(ctx, metadata); err != nil {
-		e.LogPersister.Errorf("Unabled to save approver information to deployment, %v", err)
-		return "", false
+		return false
 	}
 
 	if err := approveCmd.Report(ctx, model.CommandStatus_COMMAND_SUCCEEDED, nil, nil); err != nil {
 		e.Logger.Error("failed to report handled command", zap.Error(err))
 	}
-	return approveCmd.Commander, true
+
+	reached := e.validateApproverNum(ctx, approveCmd.Commander, num)
```

Comment:
I'm gonna say `approveCmd.Report(ctx, model.CommandStatus_COMMAND_SUCCEEDED, nil, nil)` should come after this.